src/sms/ports/inbound/events_handler.py

```diff
@@ -29,3 +30,15 @@ async def clear_topics(self, *, topics: list[str], exclude_internal: bool = True
         unless otherwise specified.
         """
         ...
+
+    @abstractmethod
+    async def publish_event(
+        self,
+        *,
+        topic: str,
+        payload: Mapping[str, str],
+        type_: bytes,
+        key: bytes,
+    ):
```

Comment:
```suggestion
        type_: str,
        key: str,
        headers: Mapping[str, str],
    ):
```
If the intended use case includes potentially supplying broken headers, then there needs to be a `headers` parameter. In the core module/actual implementation, it should then set the correlation ID conditionally.